### ISSUE TITLE
oneSignal new segments

### ISSUE THREAD

**Issue by Shimaa-Hassan** [EmanHylooz/Abjjad#91]
Not on free trial 
On free trial 
Subscribed

**Comment by Nur-alfayez** [comment]
**Group 1:** 
Not on free trial (**never** tried it) 

**Group 2:**
On free trial - need to add a time dimension to this (how many days since they clicked on start free trial? track until 30 days)

**Group 3:**
Free trial expired (did not upgrade) - need to add a time dimension to this (how many days since free trial expired? track until 30 days - and move to segment below if they end up subscribing)

**Group 4:**
Subscribed user (currently paying) - we do track how many months already, right? 

**Group 5:**
Canceled Subscription (used to be a paying user but canceled their subscription) - need to add a time dimension to this (how many days since cancelation happened? track until 30 days). 

----------------------------------------

We need to fire automated messages based on group + time + behavior, example: 

A user in free trial (group) day 3 (time) who did not download any book _(such behavior events are for next stages)_ gets an automated message saying "download your first book". 

The only behaviors we will fire automated messages based on at this stage are: 

- User started free trial 
- User upgraded (automatically through iOS or manually through Android) 
- User canceled subscription 

And if its not so hard to do one extra behavior event: 

- User on free trial finished reading a free book

**Comment by MahmoudAlyIbrahim** [comment]
@Nur-alfayez , I need the templates of the notifications that will be sent to each group of members

**Comment by Nur-alfayez** [comment]
@MahmoudAlyIbrahim what do you mean by template? OneSignal sends text based messages. Do you mean you want the exact text to be sent for each group? Because we may send different messages but its all text based following OneSignal template.

**Comment by MahmoudAlyIbrahim** [comment]
Yes, i mean text messages

**Comment by Nur-alfayez** [comment]
Can you tell me why you need them Mahmoud? Do you need to see if there are any parameters/variables to include? 

For the messages I'll have to keep testing and changing so I'll be sending them from OneSignal after you guys add the segmentations above to show in OneSignal dashboard, I need that flexibility. Let me know if there is something exact you need or if what you need is an example of the messages. 

I can send you a couple of examples by the end of tomorrow (as I'm caught up with something at the moment), would that work?

**Comment by MahmoudAlyIbrahim** [comment]
i was thinking that we will send all text messages from server, but as you need is a segment, i will refactor the code

**Comment by Nur-alfayez** [comment]
@MahmoudAlyIbrahim from what I understood during my call with @Shimaa-Hassan, we will connect the server to a tool (in our case OneSignal) that I can use to send messages from. 

@Shimaa-Hassan just looping you in to make sure I got this right :-) 

In any case, if you need us to have a call tomorrow around end of day to make sure we revisit before you guys start coding let me know.

**Comment by MahmoudAlyIbrahim** [comment]
Ok, let's have a call tomorrow

**Comment by Nur-alfayez** [comment]
3:30 pm Cairo time?

**Comment by MahmoudAlyIbrahim** [comment]
Ok :)

**Comment by Nur-alfayez** [comment]
(y)

**Comment by Shimaa-Hassan** [comment]
what about call today ?

**Comment by Shimaa-Hassan** [comment]
@Nur-alfayez

**Comment by Nur-alfayez** [comment]
Sure, let's make it at 4 Amman time.

**Comment by Shimaa-Hassan** [comment]
what about 1:00 Amman time

**Comment by Nur-alfayez** [comment]
Let's make it 1:30 if you want? I need to look at events before we talk.

**Comment by Shimaa-Hassan** [comment]
Ok great

**Comment by Nur-alfayez** [comment]
Nur to add all events/segmentations + analytics to be tracked, Shimaa to check tools and talk to OneSignal to see if its doable.

**Comment by Nur-alfayez** [comment]
**Activity Events** 

- Downloaded a (free/premium) book
- Finished % (percentage number) of a (free/premium) book 
- Added book (title) to I read it
- Added book (title) to I'll read it
- Added book (title) to I'm reading it
- Added quote for book (title) 
- Added review for book (title) 
- Clicked notify me once book (title) is available + i'll read it 
Find a way to exclude people how downloaded a book (title) on have it on any of their shelves 

Its important to mention that events must be reflected on OneSignal in a way that I can create user segments with. As filter types under user segmentation, that is the only case I would be able to create a segment that goes like this: 

If progress reading book (title) equal to (30) % AND Added quote for book (title) 
_(parameter should say that its the same book title)_ 

Can we also add another parameter (book category)? in a way where we can recommend a certain book for segments who finished books from same category? 

Also please review automation inside OneSignal, from my end I'm able to choose segments but I cannot define the time factor, for example: I need to automate sending a message to that segment 3 days after taking the event. What do we go with this? 

Does working on those events for push notification mean we'll also have these events available for tracking too? or makes it easier to add them to Appsflyer? 
Same question but for email automation.

**Comment by Nur-alfayez** [comment]
@Shimaa-Hassan, this is ready. Please let me know when you're available to discuss after you review them.

**Comment by Shimaa-Hassan** [comment]
what about call today ?

**Comment by Nur-alfayez** [comment]
Sure, how about 3 Cairo time?

**Comment by Shimaa-Hassan** [comment]
what about 12:00 cairo time ?

**Comment by Nur-alfayez** [comment]
Ok sure I'll call you

**Comment by Shimaa-Hassan** [comment]
@EmanHylooz we need a call for discuss this ?

**Comment by EmanHylooz** [comment]
@Shimaa-Hassan Yes. Let;s do it after you release the performance optimization for IOS and Android, better?

**Comment by Shimaa-Hassan** [comment]
Ok

**Comment by Shimaa-Hassan** [comment]
I will reply tomorrow ISA related to can and can't segment after oneSignal reply to my email

**Comment by Nur-alfayez** [comment]
Thanks Shimaa

**Comment by EmanHylooz** [comment]
Great!

Best,

Eman Hylooz

Founder & CEO | Abjjad

Jordan: +962.79.5402965 | Lebanon: +961.81.667918 | *www.abjjad.com*
<http://www.abjjad.com/>

**Comment by EmanHylooz** [comment]
@Shimaa-Hassan Nur is waiting for information from you about what is doable.

**Comment by Shimaa-Hassan** [comment]
@Nur-alfayez After search for what we can do with oneSignal. I found that oneSignal didn't completely support segments based on user behavior. for easy solution we can search for another tool that support segments based on user behavior or made it by own from abjjad server side.
with oneSignal all activity events will be very complex.

Search for tool that do: automated push notifications to specific segments based on user behavior

**Comment by Nur-alfayez** [comment]
Ok will do that Shimaa. Does that mean we should wait until we find another
tool to discuss if I should re-segment events based on our discussion, or
do we still need to do that?

**Comment by Shimaa-Hassan** [comment]
If u re-segment events we will change all events to fixed things.
Example:
"Downloaded a (free/premium) book (title)" changed to "Downlaoded (count) free books" and "Downlaoded (count) premium books"

**Comment by EmanHylooz** [comment]
@Nur-alfayez @Shimaa-Hassan there is a tool call push bots, the founder is Egyptian and a friend of mine. He understands our needs very well and willing to support, yet he said that for some scenarios we have it will be hard to implement (sending different parameters to the same user segment, like a specific book title for each user, based on a segmentation). 

Here is the push bots website: https://pushbots.com/
 Amr email he is willing to support: amr@pushbots.com

If you could not find a diff tool that supports everything, look into push bots as at least there we can have intensive support from Amr.

**Comment by EmanHylooz** [comment]
1. Divide the notifications into two types: from tool, from server. 
2. Shimaa found tools that support dynamic but it was very expensive @Shimaa-Hassan please send the name of the tool. 
3. Shimaa will check the documentation for push bots and tell us results enno who to use, one signal or push bots

**Comment by Nur-alfayez** [comment]
Divide the notifications into two types: from tool, from server.

@EmanHylooz should I do this? or wait until we decide what tool to use?

**Comment by EmanHylooz** [comment]
@Nur-alfayez will call yo uto give you summary :) just ended a long phone call with Shimaa about it.

**Comment by Nur-alfayez** [comment]
Ah ok cool (y)
